CMakeLists.txt

```diff
@@ -157,27 +145,10 @@ zephyr_compile_options(
 )
 
 zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:CXX>:-std=${STD_CPP_DIALECT}>
-  $<$<COMPILE_LANGUAGE:CXX>:-fcheck-new>
-  $<$<COMPILE_LANGUAGE:CXX>:-ffunction-sections>
-  $<$<COMPILE_LANGUAGE:CXX>:-fdata-sections>
```

Comment:
This line and the above is redundant with

zephyr_cc_option(-ffunction-sections -fdata-sections)

from arch/common/CMakeLists.txt

and can be safely removed, instead of ported.

(I ran some tests to confirm, the ordering of flags is changed, but the set of flags remains unchanged)

As an aside, you could move

zephyr_cc_option(-ffunction-sections -fdata-sections)

to the toplevel CMakeLists.txt if you wish.

Reply:
Fixed in separate commit.